sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -510,8 +510,13 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     requireTableExists(db, tableDefinition.identifier.table)
     verifyTableProperties(tableDefinition)
 
+    // Add table metadata such as table schema, partition columns, etc. if they aren't already
+    // present.
+    val withMetaProps = tableDefinition.copy(
```

Comment:
`alterTable` assumes we won't alter table schema(see the doc of `alterTable`), and I'm not sure if we can just put the schema into table properties and alter it.

I'd like to create a new method `def alterSchema(db: String, table: String, schema: StructType)`, which reads the raw hive table, put table schema into table properties and alter it.